lib/urn.js

```diff
@@ -44,7 +44,77 @@ module.exports = {
       if (!urn.match(allowedProtocolsRegExp) || !urn.match(/^([^:]+:)+[^:]+$/)) {
         return false;
       }
+
+      // Parse ID portion of URN
+      var cat = urn.split(':').slice(0, -1)
+      var id = urn.split(':').slice(-1)[0];
+
+      // Split identity by each byte
+      var charArr = id.match(/.{1,2}/g);
```

Comment:
- length of chars should alway be even, if it's not then should not continue
- from this perspective should be {2}
- a preferable way is to use dedicated type Buffer for this kind of operations (though it's not required)
- would be nice to use more specific var name